aten/src/ATen/TensorBase.h

```diff
@@ -48,6 +51,9 @@ struct TensorBase {
   TensorImpl * get() const {
     return pImpl;
   }
+  bool defined() const {
+    return pImpl != UndefinedTensor::singleton();
```

Comment:
This will be a functional call which can't be inlined. It's used in every `checked_cast` operation, so it'd be good to make it as cheap as possible.

You can avoid the call if you declare the  `extern TensorImpl*` singleton in an included header and use that (or make `UndefinedTensor::singleton` inline in a header)

Reply:
Now that undefined is always a valid tensor, can we simply skip or reorder the `defined` called as well? In general we might be able to remove some `defined()` now when we are only doing it to prevent a null dereference.

Reply:
The checked_cast code doesn't actually call defined() -- it just checks that the expr is non-null.  So that should be able to go away.